assembl/static2/js/app/components/debate/brightMirror/sideComment/innerBoxView.jsx

```diff
@@ -4,71 +4,120 @@ import React, { Fragment } from 'react';
   comment: ExtractComment,
-  changeCurrentExtract: (?number) => void
+  changeCurrentExtract: (?number) => void,
+  setEditMode: ?(number, string) => void
 };
 
-const renderRichtext = (text: string) => activeHtml(text && transformLinksInHtml(text), postBodyReplacementComponents());
+export type State = {
+  menuTarget: HTMLElement | null
+};
+
+class InnerBoxView extends React.Component<Props, State> {
+  state = {
+    menuTarget: null
+  };
+
+  renderRichtext = (text: string) => activeHtml(text && transformLinksInHtml(text), postBodyReplacementComponents());
 
-const InnerBoxView = ({ contentLocale, extractIndex, extracts, comment, changeCurrentExtract }: Props) => {
-  const displayName =
-    comment && comment.creator && !comment.creator.isDeleted ? comment.creator.displayName : I18n.t('deletedUser');
+  render() {
+    const { contentLocale, extractIndex, extracts, comment, changeCurrentExtract, setEditMode } = this.props;
+    const { menuTarget } = this.state;
+    const displayName =
+      comment && comment.creator && !comment.creator.isDeleted ? comment.creator.displayName : I18n.t('deletedUser');
+    const canEdit =
+      comment &&
+      comment.creator &&
+      String(comment.creator.userId) === getConnectedUserId() &&
+      connectedUserCan(Permissions.EDIT_MY_POST) &&
+      // Prevent editing reply for now (will be added in another ticket)
+      setEditMode;
 
-  return (
-    <Fragment>
-      <div className="harvesting-box-header">
-        <div className="profile">
-          <AvatarImage userId={comment.creator.userId} userName={displayName} />
-          <div className="harvesting-infos">
-            <div className="username">{displayName}</div>
-            <div className="harvesting-date" title={comment.creationDate}>
-              {moment(comment.creationDate)
-                .locale(contentLocale)
-                .fromNow()}
+    return (
+      <Fragment>
+        <div className="harvesting-box-header">
+          <div className="profile">
+            <AvatarImage userId={comment.creator.userId} userName={displayName} />
+            <div className="harvesting-infos">
+              <div className="username">{displayName}</div>
+              <div className="harvesting-date" title={comment.creationDate}>
+                {moment(comment.creationDate)
+                  .locale(contentLocale)
+                  .fromNow()}
+              </div>
             </div>
-          </div>
-        </div>
-      </div>
-      <div className="harvesting-box-body">
-        <div className="body-container">
-          <div className="previous-extract">
-            {extracts &&
-              extractIndex > 0 && (
-                <div
-                  onClick={() => {
-                    changeCurrentExtract(-1);
+            {canEdit && (
+              <div className="button-bar">
+                <Button
+                  className="action-menu-btn"
+                  onClick={(e) => {
```

Comment:
add a method for the onClick event: onClick={this.blabla}

Reply:
I didn't do this because there was a rendering problem when the onClick is activated at render so I had to use this format (e) => {...} and then there is no need to create a function with 1 line of code in addition